pyfive/h5d.py

```diff
@@ -84,11 +84,18 @@ def __init__(self, dataobject, pseudo_chunking_size_MB=4):
         self._msg_offset, self.layout_class,self.property_offset = dataobject.get_id_storage_params()
         self._unique = (self._filename, self.shape, self._msg_offset)
 
+        #FIXME: Everyone of these parses the message each time. Optimisation possible?
```

Comment:
Yes, maybe just initialize it once either in the first `if` or before?

Reply:
It's a problem in lots of places, but I think i fixed it here.